pkg/util/validation/limits.go

```diff
@@ -203,6 +203,7 @@ type Limits struct {
 	CardinalityAnalysisEnabled                    bool `yaml:"cardinality_analysis_enabled" json:"cardinality_analysis_enabled"`
 	LabelNamesAndValuesResultsMaxSizeBytes        int  `yaml:"label_names_and_values_results_max_size_bytes" json:"label_names_and_values_results_max_size_bytes"`
 	LabelValuesMaxCardinalityLabelNamesPerRequest int  `yaml:"label_values_max_cardinality_label_names_per_request" json:"label_values_max_cardinality_label_names_per_request"`
+	CardinalityAPIMaxSeriesLimit                  int  `yaml:"cardinality_api_max_series_limit" json:"cardinality_api_max_series_limit"`
```

Comment:
The limit isn't really limiting the number of series returned. Can you change the setting (and CLI flag, and methods) to `cardinality_analysis_max_results`.